src/libraries/Microsoft.PowerFx.Core/Entities/External/DataSourceInfo.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Entities.Delegation;
+using Microsoft.PowerFx.Core.Entities.QueryOptions;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Functions.Delegation.DelegationMetadata;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    // Implements a base data source, used in DType.AssociatedDataSources, itself in RecordType constructor to host a CDP record type
+    internal class DataSourceInfo : IExternalTabularDataSource
+    {
+        public const string IsChoiceValue = "Value";
```

Comment:
What is this for?

Reply:
Same as CapabilityConstants.IsChoiceValue so we can remove that one
It's not used 
Even the one in CapabilityConstants isn't as I can't find anyone explaining me what "x-ms-sp": { IsChoice: bool } is really doing (and why?)
It seems transforming a "Choice" property to "Choice.Value" when set to true, per TestSharepointODataColumnCapabilities test in PowerApps code but that's all I could find + I can't find any real example (CDP Get Schema with "IsChoice") + I can't get access to SP code despite multiple tries...